pkg/controller/node/controller_utils.go

```diff
@@ -290,31 +290,35 @@ func recordNodeStatusChange(recorder record.EventRecorder, node *v1.Node, new_st
 }
 
 // Returns true in case of success and false otherwise
-func swapNodeControllerTaint(kubeClient clientset.Interface, taintToAdd, taintToRemove *v1.Taint, node *v1.Node) bool {
-	taintToAdd.TimeAdded = metav1.Now()
-	err := controller.AddOrUpdateTaintOnNode(kubeClient, node.Name, taintToAdd)
+func swapNodeControllerTaint(kubeClient clientset.Interface, taintsToAdd, taintsToRemove []*v1.Taint, node *v1.Node) bool {
+	for _, taintToAdd := range taintsToAdd {
+		taintToAdd.TimeAdded = metav1.Now()
+	}
+
+	err := controller.AddOrUpdateTaintOnNode(kubeClient, node.Name, taintsToAdd...)
 	if err != nil {
 		utilruntime.HandleError(
 			fmt.Errorf(
 				"unable to taint %v unresponsive Node %q: %v",
-				taintToAdd.Key,
+				taintsToAdd,
```

Comment:
this will print pointer addresses. Make sure that it prints something understandable.